src/Diagnostics.RuntimeHost/Services/GithubClient/AzureStorageSourceCodeClient.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Diagnostics.RuntimeHost.Models;
+using Diagnostics.RuntimeHost.Services.StorageService;
+using Diagnostics.RuntimeHost.Utilities;
+using Octokit;
+
+namespace Diagnostics.RuntimeHost.Services
+{
+    /// <summary>
+    /// A temporary azure storage implementation of the IGitHubClient library.
+    /// </summary>
+    public sealed class AzureStorageSourceCodeClient : IGithubClient
+    {
+        private IStorageService _storageService;
+
+        public string UserName => string.Empty;
+
+        public string RepoName => string.Empty;
+
+        public string Branch => string.Empty;
+
+        public AzureStorageSourceCodeClient(IStorageService storageService)
+        {
+            _storageService = storageService;
+        }
+
+        public Task CreateOrUpdateFile(string destinationFilePath, string content, string commitMessage, bool convertContentToBase64 = true)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task CreateOrUpdateFiles(IEnumerable<CommitContent> commits, string commitMessage)
+        {
+            if (commits.Any(content => content.FilePath.Contains("kustoClusterMappings", StringComparison.CurrentCultureIgnoreCase)))
+            {
+                //no op
+                return;
+            }
+
+            foreach (var commit in commits)
+            {
+                if (!Equals(commit.EncodingType, EncodingType.Base64))
+                {
+                    commit.Content = InternalAPIHelper.Base64Encode(commit.Content);
+                }
+                await _storageService.LoadBlobToContainer(commit.FilePath, commit.Content).ConfigureAwait(false);
+            }
```

Comment:
We should save the metadata to table here

Reply:
https://github.com/Azure/Azure-AppServices-Diagnostics/blob/3e2c1c450407e27fa61f876ddd473e987c12d78d/src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs#L135

Reply:
I think your code is already updating the table

Reply:
So on top of the regular updates to blob and table, you are saving each file (like metadata.json, package.json) to the blob container?

Reply:
Yeah specifically the detector and gist file and package.json

Reply:
You can filter out dll and pdb here. Dll is saved as part of StorageWatcher and pdb is not required